### PR TITLE
Updated load_data.R #1592

### PR DESCRIPTION
**Pull request by moki1202**
removed library functions by appropriately providing namespaces to functions.

### CODE REVIEW

**Comment by ashiklom** [comment]
Thanks @moki1202 ! This looks great!

FYI, the checks were failing because your changes call for an update to the automated package documentation (specifically, the `modules/benchmark/NAMESPACE` file needs to be updated to reflect the new `@importFrom magrittr %>%). Typically, this means you have to run `devtools::document()` from inside the package and then committing the resulting changes.

However, no need to do that here because we have a GitHub Actions bot that will do it for you! My cryptic comment above activated our documentation bot, which will automatically run `devtools::document()` for all the PEcAn packages and commit any changes.

**Review by mdietze — APPROVED** [review]

**Review by infotroph — APPROVED** [review]

File: modules/benchmark/DESCRIPTION [hunk 1]
```diff
@@ -35,6 +35,7 @@ Imports:
 Suggests:
     PEcAn.data.land,
     testthat (>= 2.0.0)
```
Comment (ashiklom) [inline modules/benchmark/DESCRIPTION:37]: Need a comma here.

```suggestion
    testthat (>= 2.0.0),
```

File: modules/benchmark/DESCRIPTION [hunk 2]
```diff
@@ -35,6 +35,7 @@ Imports:
...
+    bd
```
Comment (ashiklom) [inline modules/benchmark/DESCRIPTION:38]: `bd` was the variable used for the name of the BrownDog package. This should be the actual name of the package:

```suggestion
    BrownDog
```

Comment (moki1202) [reply modules/benchmark/DESCRIPTION:38]: @ashiklom sorry for the above mistake. Had a good read about the significance of suggests section in DESCRIPTION.

File: modules/benchmark/R/load_data.R [hunk 3]
```diff
@@ -30,12 +31,7 @@ load_data <- function(data.path, format, start_year = NA, end_year = NA, site =
     vars.used.index <- setdiff(seq_along(format$vars$variable_id), format$time.row)
   }
   
-  library(PEcAn.utils)
-  library(PEcAn.benchmark)
-  library(lubridate)
-  library(udunits2)
-  library(dplyr)
-  
+
```
Comment (infotroph) [inline modules/benchmark/R/load_data.R:34]:
```suggestion
```

**Comment by infotroph** [comment]
@moki1202 Can you please add `magrittr,` to the Imports section of the DESCRIPTION file, between current lines 23 and 24? I'd submit it as a suggested change but it's too far from any changed lines.

Extra credit if you want to bother reordering the existing import declarations so the whole listing is actually alphabetical, but clearly the rest of us haven't done that...  🙈

File: modules/benchmark/DESCRIPTION [hunk 4]
```diff
@@ -15,26 +15,29 @@ Description: The Predictive Ecosystem Carbon Analyzer (PEcAn) is a scientific
...
     stringr
+    udunits2 (>= 0.11),
+    XML (>= 3.98-1.4),
+    zoo,  
+
 Suggests:
     PEcAn.data.land,
-    testthat (>= 2.0.0)
+    testthat (>= 2.0.0),
+    BrownDog,
```
Comment (ashiklom) [inline modules/benchmark/DESCRIPTION:40]: Trailing commas are not allowed in `DESCRIPTION` files.

```suggestion
    BrownDog
```

File: modules/benchmark/DESCRIPTION [hunk 5]
```diff
@@ -15,26 +15,29 @@ Description: The Predictive Ecosystem Carbon Analyzer (PEcAn) is a scientific
...
-    ggplot2,
-    gridExtra,
     reshape2,
-    dbplyr,
     SimilarityMeasures,
-    zoo,
     stringr
+    udunits2 (>= 0.11),
+    XML (>= 3.98-1.4),
+    zoo,
```
Comment (ashiklom) [inline modules/benchmark/DESCRIPTION:35]:
```suggestion
    zoo
```

File: modules/benchmark/DESCRIPTION [hunk 6]
```diff
@@ -15,26 +15,29 @@ Description: The Predictive Ecosystem Carbon Analyzer (PEcAn) is a scientific
...
-    udunits2 (>= 0.11),
-    XML (>= 3.98-1.4),
-    dplyr,
-    ggplot2,
-    gridExtra,
     reshape2,
-    dbplyr,
     SimilarityMeasures,
-    zoo,
     stringr
```
Comment (ashiklom) [inline modules/benchmark/DESCRIPTION:32]: You _do_ need the comma here though. This was must have been accidentally dropped when you re-alphabetized.

```suggestion
    stringr,
```

File: modules/benchmark/DESCRIPTION [hunk 7]
```diff
@@ -15,26 +15,29 @@ Description: The Predictive Ecosystem Carbon Analyzer (PEcAn) is a scientific
...
     reshape2,
-    dbplyr,
     SimilarityMeasures,
-    zoo,
-    stringr
+    stringr,
+    udunits2 (>= 0.11),
+    XML (>= 3.98-1.4),
+    zoo 
+
```
Comment (ashiklom) [inline modules/benchmark/DESCRIPTION:36]:
```suggestion
```

File: modules/benchmark/R/load_data.R [hunk 8]
```diff
@@ -44,13 +39,13 @@ load_data <- function(data.path, format, start_year = NA, end_year = NA, site =
     fcn <- match.fun(fcn1)
   } else if (exists(fcn2)) {
     fcn <- match.fun(fcn2)
-  } else if (!exists(fcn1) & !exists(fcn2) & require(bd)) { 
+  } else if (!exists(fcn1) & !exists(fcn2) & requireNamespace(bd, quietly = TRUE)) {
```
Comment (infotroph) [inline modules/benchmark/R/load_data.R:42]:
```suggestion
  } else if (!exists(fcn1) & !exists(fcn2) & requireNamespace("BrownDog", quietly = TRUE)) { 
```

File: modules/benchmark/DESCRIPTION [hunk 9]
```diff
@@ -15,26 +15,28 @@ Description: The Predictive Ecosystem Carbon Analyzer (PEcAn) is a scientific
...
-    XML (>= 3.98-1.4),
-    dplyr,
-    ggplot2,
-    gridExtra,
     reshape2,
-    dbplyr,
     SimilarityMeasures,
-    zoo,
-    stringr
+    stringr,
```
Comment (infotroph) [inline modules/benchmark/DESCRIPTION:32]:
```suggestion
    stringr,
    tidyselect,
```

**Review by infotroph — CHANGES_REQUESTED** [review]
Hopefully last tweaks. Thanks for sticking with this through so many rounds of 'one more thing' -- clearly this code was long overdue for a cleanup!

File: modules/benchmark/DESCRIPTION (hunk 9 above)
Comment (moki1202) [reply modules/benchmark/DESCRIPTION:32]: @infotroph ill get on this ASAP. also, the pleasure was mine, got to learn so much from you guys. thanks for being patient with me hehe.

Comment (moki1202) [reply modules/benchmark/DESCRIPTION:32]: also please suggest any error fix that I can start working on after this.

**Review by ashiklom — APPROVED** [review]

**Review by infotroph — APPROVED** [review]

**Pull request merged**